### PR TITLE
Create endpoint for MVI proxy add

### PR DESCRIPTION
**Pull request by SarahJaine**
## Description of change

Create new endpoint to make orchestrated search call to MVI to add users who are missing `participant_id` and `birls_id` or just `participant_id`.  This functionality was previously a before action on the `in_progress_forms` endpoint.  For reasons detailed in the attached ticket, this functionality needed to be moved up in the user flow.  The plan now if for the FE to call this new endpoint before the `intent_to_file` endpoint if the user has `ADD_PERSON` in the services list.

## Original issue(s)
First PR for department-of-veterans-affairs/va.gov-team#9856
Second PR for this issue is https://github.com/department-of-veterans-affairs/vets-api/pull/4390

## Things to know about this PR

**Feedback requested:**
- I'm not at all attached to the name of this endpoint, it's kind of stupid.  Can you think of something more apt?
- Check out all my comments on specific lines of code below

Changes
- [x] remove proxy add from `in_progress_forms` endpoint
- [x] removes proxy add description from  swagger docs for `in_progress_forms` endpoint
- [x] create new swagger docs for this endpoint
- [x] moves/renames evss `access_original_claims?` policy to mvi `access_add_person?` policy 
- [x] write new tests for `access_add_person?` policy
- [x] rename `ORIGINAL_CLAIMS` attribute in services list

### CODE REVIEW

File: app/controllers/v0/mvi_users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          detail: 'No birls_id while participant_id present',
+          source: 'MviUsersController'
+        )
+      end
+
+      # Add user to MVI
+      add_response = @current_user.mvi.mvi_add_person
+      raise add_response.error unless add_response.ok?
+
+      render json: { "message": add_response }
```
Comment (SarahJaine) [inline app/controllers/v0/mvi_users_controller.rb:29]: This renders the following message value:
```
<MVI::Responses::AddPersonResponse:0x00007ff4be09e830 @status="OK",
@mvi_codes={:birls_id=>"111985523", :participant_id=>"32397028"}, @error=nil>
```
It seems traditional to return some sort of value for a POST request. However, I'm not sure it's appropriate/necessary to return user ids. Thoughts on what we should return here?

File: config/routes.rb [hunk 2]
```diff
@@ -44,6 +44,10 @@
       get 'user_submissions'
     end
 
+    resource :mvi_users, only: [] do
+      post ':id', to: 'mvi_users#submit'
+    end
```
Comment (SarahJaine) [inline config/routes.rb:49]: This seems really verbose... is there a better way to do this?

File: spec/request/mvi_users_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          sign_in_as(user)
+        end
+
+        it 'return 200, add user to MVI' do
+          VCR.use_cassette('mvi/add_person/add_person_success') do
+            VCR.use_cassette('mvi/find_candidate/orch_search_with_attributes') do
+              # expect success to be achieved by calling mvi_add_person
+              expect_any_instance_of(Mvi).to receive(:mvi_add_person).once.and_call_original
+              post "/v0/mvi_users/#{valid_form_id}"
+              expect(response.status).to eq(200)
```
Comment (SarahJaine) [inline spec/request/mvi_users_request_spec.rb:70]: This test is copied straight from the proxy add tests when this functionality was part of the in progress forms endpoint.  I'm not super happy that we're just testing that `mvi_add_person` is called but this was the best I could do.  Happy to hear any thoughts on improvements.

File: app/controllers/v0/mvi_users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+module V0
+  class MviUsersController < ApplicationController
+    before_action { authorize :mvi, :missing_critical_ids? }
+
+    def submit
+      # Caller must be using proxy add in order to complete Intent to File or Disability Compensation forms
+      form_id = params[:id]
+      if [FormProfiles::VA0996::FORM_ID, VA526ez::FORM_ID].exclude?(form_id)
+        raise Common::Exceptions::Forbidden.new(
```
Comment (SarahJaine) [inline app/controllers/v0/mvi_users_controller.rb:11]: Do you think throwing a 403 for two scenarios: `missing_critical_ids?` policy failure and giving the wrong (form) `id` param is confusing?  The latter gets a pretty error message but still...

File: app/controllers/v0/mvi_users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+      # Scenario indicates serious problems with user data
+      if @current_user.birls_id.nil? && @current_user.participant_id.present?
+        raise Common::Exceptions::UnprocessableEntity.new(
+          detail: 'No birls_id while participant_id present',
+          source: 'MviUsersController'
+        )
+      end
+
+      # Add user to MVI
```
Comment (annaswims) [inline app/controllers/v0/mvi_users_controller.rb:25]: So, in this scenario, the person may already be in MVI, but just have id's for other sytsems I think it'd be nice to make that clear in this comment.

File: spec/request/mvi_users_request_spec.rb (hunk 3 above)
Comment (annaswims) [reply spec/request/mvi_users_request_spec.rb:70]: you'll need to add a test to swagger_spec.rb and that will validate that the response matches the docs.  That should round this out.

File: app/policies/mvi_policy.rb [hunk 6]
```diff
@@ -1,12 +1,32 @@
 # frozen_string_literal: true
 
+# rubocop:disable Metrics/BlockLength
+
 MviPolicy = Struct.new(:user, :mvi) do
+  def access_add_person?
+    if user.edipi.present? && user.ssn.present? && (user.birls_id.blank? || user.participant_id.blank?)
+      log_success
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply app/policies/mvi_policy.rb:8]: Fair enough.

File: spec/request/swagger_spec.rb [hunk 7]
```diff
@@ -661,6 +661,19 @@
...
 
+    describe 'MVI Users' do
+      # let(:mhv_user) { build(:user_with_no_ids) }  # returns 404, The record for the requested user could not be found
+      # let(:mhv_user) { build(:user, :loa1) }   # returns 403, Forbidden User does not have access to the requested resource
+
+      it 'supports creating mvi user' do
+        expect(subject).to validate(:post, '/v0/mvi_users/{id}', 401, 'id' => '21-0966')
+        VCR.use_cassette('mvi/add_person/add_person_success') do
+          byebug
+          expect(subject).to validate(:post, '/v0/mvi_users/{id}', 200, headers.merge('id' => '21-0966'))
```
Comment (SarahJaine) [inline spec/request/swagger_spec.rb:672]: This is killing me. How does this work?

File: app/controllers/v0/mvi_users_controller.rb (hunk 5 above)
Comment (SarahJaine) [reply app/controllers/v0/mvi_users_controller.rb:25]: @annaswims Are you just asking me to be more specific? E.g. write a one shorter version of the swagger description >> https://github.com/department-of-veterans-affairs/vets-api/pull/4384/files#diff-f2e2eff8088c7aa0ac8d81ead95df70aR16-R21

File: spec/request/swagger_spec.rb (hunk 7 above)
Comment (SarahJaine) [reply spec/request/swagger_spec.rb]: Fixed with Anna's help! 🎉

**Comment by SarahJaine** [comment]
Build is failing because branch was updated from master and pulled in breaking security scan

**Comment by erluti** [comment]
> Build is failing because branch was updated from master and pulled in breaking security scan

Merge in the latest, a fix has been merged https://github.com/department-of-veterans-affairs/vets-api/pull/4391

**Pull request merged**